### PR TITLE
Fix amp + device_map

### PR DESCRIPTION
**Pull request by tastelikefeet**
1. Fix the amp + device_map bug in chatglm2 finetune code 
2. Optional to save optimizer state 
3. Fix the logger double print problem

### CODE REVIEW

File: examples/pytorch/chatglm6b/finetune.py [hunk 1]
```diff
@@ -228,7 +243,10 @@ def cfg_modify_fn(cfg):
         rank=args.lora_rank,
         lora_alpha=args.lora_alpha,
         lora_dropout=args.lora_dropout)
-    model = model.bfloat16()
+    if args.use_amp:
+        model = model.float()
```
Comment (wenmengzhou) [inline examples/pytorch/chatglm6b/finetune.py:247]: will changing model to float  affect the accuracy of the finetuned model?

File: modelscope/trainers/hooks/checkpoint/checkpoint_processor.py [hunk 2]
```diff
@@ -175,20 +178,22 @@ def link(self, model, src_file, output_dir):
...
             meta: Some extra meta info.
+            save_optimizers: Save optimizers state or not.
         """
         save_checkpoint(
             model,
             train_state_file,
-            trainer.optimizer,
-            trainer.lr_scheduler,
+            trainer.optimizer if save_optimizers else None,
+            trainer.lr_scheduler if save_optimizers else None,
```
Comment (wenmengzhou) [inline modelscope/trainers/hooks/checkpoint/checkpoint_processor.py:196]: It appears that, apart from the optimizer and learning rate (lr), other information does not need to be periodically saved.

File: examples/pytorch/chatglm6b/finetune.py (hunk 1 above)
Comment (tastelikefeet) [reply examples/pytorch/chatglm6b/finetune.py:247]: amp will do the mixed precision thing, so if args.use_amp is True, the amp hook will be added
